crates/spin-js-engine/src/lib.rs

```diff
@@ -635,10 +637,16 @@ fn open_kv(context: &Context, _this: &Value, args: &[Value]) -> Result<Value> {
                     move |context, _this: &Value, args: &[Value]| match args {
                         [key] => {
                             let key = deserialize_helper(key)?;
-                            let value = store.get(key)?;
-                            let mut serializer = Serializer::from_context(context)?;
-                            value.serialize(&mut serializer)?;
-                            Ok(serializer.value)
+                            let result = store.get(key);
+                            match result {
+                                Ok(val) => {
+                                    let mut serializer = Serializer::from_context(context)?;
+                                    val.serialize(&mut serializer)?;
+                                    Ok(serializer.value)
+                                }
+                                Err(key_value::Error::NoSuchKey) => context.null_value(),
+                                Err(e) => bail!(e),
```

Comment:
`bail` might do some kind of unexpected conversion here -- could we change it to:
```suggestion
                                Err(e) => return Err(e.into()),
```

Reply:
Ah! Will change it.